File: 2102-find-the-middle-index-in-array/find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        leftsum = 0
        total = sum(nums)
        for left in range(len(nums)):
            if leftsum == total - leftsum - nums[left]:
                return left
            leftsum += nums[left]
        return -1 
        

File: 2102-find-the-middle-index-in-array/find-the-middle-index-in-array.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        int total = 0;
        for(int num: nums) {
            total += num;
        }
        int sumleft = 0;
        for(int l = 0; l < nums.length; l++) {
            if (sumleft != total - sumleft - nums[l]) {
                sumleft += nums[l];
            }
            else {
                return l;
            }
        }
        return -1;
    }
}